packages/amplify-migration-codegen-e2e/src/assertions.ts

```diff
@@ -79,15 +138,35 @@ export async function assertUserPoolResource(projRoot: string, gen1UserPoolId: s
 
 export async function assertStorageResource(projRoot: string, gen1BucketName: string, gen1Region: string) {
   const gen1Resource = await getResourceDetails('AWS::S3::Bucket', gen1BucketName, gen1Region);
-  removeProperties(gen1Resource, ['DualStackDomainName', 'DomainName', 'BucketName', 'Arn', 'RegionalDomainName', 'Tags', 'WebsiteURL']);
+  removeProperties(gen1Resource, [
+    'DualStackDomainName',
+    'DomainName',
+    'BucketName',
+    'Arn',
+    'RegionalDomainName',
+    'Tags',
+    'WebsiteURL',
+    'NotificationConfiguration.LambdaConfigurations[0].Function',
+    'NotificationConfiguration.LambdaConfigurations[1].Function',
```

Comment:
We should add a validation that the `*.Function` field in gen 1 and gen 2 stacks should be both empty or non-empty.

Reply:
do the same for assertion places where the function name is excluded.

Reply:
Yes, that makes sense. Will do.

Reply:
Addressed. [dbf44b4](https://github.com/aws-amplify/amplify-cli/pull/13978/commits/dbf44b417f56692be3608d684dc8f31f985b6df8)